lib/parsers/aff_parser.rb

```diff
@@ -0,0 +1,38 @@
+module Parsers
+  class AffParser
+    AFFIX_REGEX = /^(PFX|SFX)(\s|\w|.)*/
+    AFFIX_GROUP_REGEX = /^(PFX|SFX)\s\w/
+
+    def initialize(file)
+      @file = file
+    end
+
+    def parse
+      {}.tap do |res|
+        res[:affixes] = @file.to_a
+                             .map { |ln| ln.tr("\n\t", '') }
+                             .grep(AFFIX_REGEX)
+                             .group_by { |el| el[AFFIX_GROUP_REGEX] }.values
+                             .map(&method(:parse_affix_line))
+      end
+    end
+
+    private
+
+    def parse_affix_line(aff_group)
+      header = aff_group.first
```

Comment:
You can do `header, *rule_lines = aff_group`, then you'll not need ugly `[1..-1]` below.